### DISCUSSION TITLE
Simple static methods for request validation

### DISCUSSION THREAD

**Discussion by dougkusanagi** [laravel/framework#50222] [Ideas]
Hello pals, I was thinking it would interesting have a class with all common validation rules to get auto-completions in request validations. I know Spatie has a package for validations, but i had some simpler idea in my mind, i don't know if would be helpfull or if Laravel already has it, so i'm posting here.

This is some snippet of what i would like to be able to do:

![image](https://github.com/laravel/framework/assets/4421025/f8e6e1ee-54ca-4783-9c29-c0c29c6e1cc2)

```php
enum RulesEnum: string {
    case Required = 'required';
    case Str = 'string';
    case Lowercase = 'lowercase';
    case Max = 'max';
}

class Rules {
    public static function required(): RulesEnum {
        return RulesEnum::Required;
    }
    public static function string(): RulesEnum {
        return RulesEnum::Str;
    }
    public static function lowercase(): RulesEnum {
        return RulesEnum::Lowercase;
    }
    public static function max(int $value): String {
        return RulesEnum::Max . ":$value";
    }
}

class ProfileUpdateRequest extends FormRequest {
    public function rules(): array {
        return [
            // 'name' => ['required', 'string', 'max:255'],
            'name' => [Rules::required(), Rules::string(), Rules::max(255)],
        ];
    }
}
```

**Comment by henzeb** [comment]
instead of making something new, extend something old: \Illuminate\Validation\Rule...

But i don't know, I like the string based rules. Especially when they have no purpose of being an object. `Rule::string() has more characters than just `string`.

**Reply by dougkusanagi** [reply]
Strings could be also acepted, but can search by a certain rule in autocomplete would be also usefull, i think.

**Reply by henzeb** [reply]
I don't know. Maybe you should take a look at laravel-idea, if you are developing in PhpStorm?
